src/jmespath.js

```diff
@@ -1131,6 +1135,278 @@
+                  if (isScalar(value[node.name])) value[node.name] = fn(value[node.name]); // Over.
+                  field = value[node.name];
+                  if (field === undefined) {
+                      return null;
+                  } else {
+                      return field;
+                  }
+              }
+              return null;
+            case "Subexpression":
+              result = this.visit(node.children[0], value, fn);
+              for (i = 1; i < node.children.length; i++) {
+                  result = this.visit(node.children[1], result, fn);
+                  if (result === null) {
+                      return null;
+                  }
+              }
+              return result;
+            case "IndexExpression":
+              left = this.visit(node.children[0], value, fn);
+              right = this.visit(node.children[1], left, fn);
+              return right;
+            case "Index":
+              if (!isArray(value)) {
+                return null;
+              }
+              var index = node.value;
+              if (index < 0) {
+                index = value.length + index;
+              }
+              if (isScalar(value[index])) value[index] = fn(value[index]); // Over.
+              result = value[index];
+              if (result === undefined) {
+                result = null;
+              }
+              return result;
+            case "Slice":
+              if (!isArray(value)) {
+                return null;
+              }
+              var sliceParams = node.children.slice(0);
+              var computed = this.computeSliceParams(value.length, sliceParams);
+              var start = computed[0];
+              var stop = computed[1];
+              var step = computed[2];
+              result = [];
+              if (step > 0) {
+                  for (i = start; i < stop; i += step) {
+                      if (isScalar(value[i])) value[i] = fn(value[i]); // Over.
+                      result.push(value[i]);
+                  }
+              } else {
+                  for (i = start; i > stop; i += step) {
+                      if (isScalar(value[i]))value[i] = fn(value[i]); // Over.
+                      result.push(value[i]);
+                  }
+              }
+              return result;
+            case "Projection":
+              // Evaluate left child.
+              var base = this.visit(node.children[0], value, fn);
+              if (!isArray(base)) {
+                return null;
+              }
+              collected = [];
+              for (i = 0; i < base.length; i++) {
+                current = this.visit(node.children[1], base[i], fn);
+                if (current !== null) {
+                  collected.push(current);
+                }
+              }
+              return collected;
+            case "ValueProjection":
+              // Evaluate left child.
+              base = this.visit(node.children[0], value, fn);
```

Comment:
I'm assuming this isn't `var base` because it's only called after the `Projection` case above where `base` is initially declared. Is that right?

Reply:
The code came from the original JmesPath code, I duplicated and tweaked to implement `over`. This is not one of those tweaks so I am inclined to leave it.